src/unix/os390.c

```diff
@@ -628,9 +645,20 @@ static int os390_message_queue_handler(uv__os390_epoll* ep) {
   events = 0;
   if (msg.__rfim_event == _RFIM_ATTR || msg.__rfim_event == _RFIM_WRITE)
     events = UV_CHANGE;
-  else if (msg.__rfim_event == _RFIM_RENAME)
+  else if (msg.__rfim_event == _RFIM_RENAME || msg.__rfim_event == _RFIM_UNLINK)
+    events = UV_RENAME;
+  else if (msg.__rfim_event == 156) {
+    /* TODO(gabylb): zos - this event should not happen, need to investigate.
+     *
+     * This event seems to occur when the watched file is [re]moved, or an
+     * editor (like vim) renames then creates the file on save (for vim, that's
+     * when backupcopy=no|auto), which causes __w_pioctl to fail as the file
+     * wasn't yet created; the sleep is to help resolve the rename-create issue.
+     */
+    struct timespec timeout = { 0, 1000000 }; /* 1ms. */
     events = UV_RENAME;
-  else
+    nanosleep(&timeout, NULL);
```

Comment:
This seems like it could cause somewhat severe thread starvation issues and bad data races, if someone has registered multiple paths. Calling it a `UV_RENAME` seems fine, but the nanosleep needs to be short and go with the `_IOCC_REGFILEINT` call retry loop (or wherever the actual failure occurs).

Reply:
I did some further testing and was able to get it to work without the nanosleep, so I went ahead and removed it entirely.